app/code/Magento/Customer/Controller/Adminhtml/Address/Delete.php

```diff
@@ -71,7 +71,7 @@ public function execute(): Json
         $addressId = $this->getRequest()->getParam('id', false);
         $error = false;
         $message = '';
-        if ($addressId && $this->addressRepository->getById($addressId)->getCustomerId() === $customerId) {
+        if ($addressId && $this->addressRepository->getById($addressId)->getCustomerId() === (int)$customerId) {
```

Comment:
it's better to convert all the types to single type.
```suggestion
        if ($addressId && (int)$this->addressRepository->getById($addressId)->getCustomerId() === (int)$customerId) {
```